lib/promscrape/client.go

```diff
@@ -349,13 +353,14 @@ var (
 	scrapeRetries         = metrics.NewCounter(`vm_promscrape_scrape_retries_total`)
 )
 
-func doRequestWithPossibleRetry(hc *fasthttp.HostClient, req *fasthttp.Request, resp *fasthttp.Response, deadline time.Time) error {
+func doRequestWithPossibleRetry(ctx context.Context, hc *fasthttp.HostClient, req *fasthttp.Request, resp *fasthttp.Response) error {
```

Comment:
i think, the behaviour of this function has changed...
before: the deadline was explicit, you can't call it without specifying some deadline
now: it depends on the context how the func would behave. Caller could have set deadline, or it could just use any other context it already had. No decision for specifying deadline anymore.

I'd prefer keeping deadline argument and extend the `ctx` right in function, instead of inheriting the context.

Reply:
Implemented suggested approach